src/signals-unix.c

```diff
@@ -700,12 +707,97 @@ void trigger_profile_peek(void)
         }
     }
     bt_size_cur = 0; // clear profile buffer
-    if (jl_profile_start_timer() < 0)
+    if (jl_profile_start_timer(0) < 0)
         jl_safe_printf("ERROR: Could not start profile timer\n");
     else
         profile_autostop_time = jl_hrtime() + (profile_peek_duration * 1e9);
 }
 
+void jl_profile_task_unix(size_t nthreads)
+{
+    if (jl_profile_is_buffer_full()) {
+        // Buffer full: Delete the timer
+        jl_profile_stop_timer();
+        return;
+    }
+
+    jl_task_t *t = jl_get_random_task();
+    assert(t == NULL || jl_is_task(t));
+    if (t == NULL) {
+        return;
+    }
+    int t_state = jl_atomic_load_relaxed(&t->_state);
+    if (t_state == JL_TASK_STATE_DONE) {
+        return;
+    }
+
+    jl_rec_backtrace(t);
+
+    // store threadid but add 1 as 0 is preserved to indicate end of block
+    bt_data_prof[bt_size_cur++].uintptr = 1; // dummy value for now... Is this ever used when outputting the profile?
```

Comment:
definitely used to output the profile by `Profile.print` when used with `groupby`:
```
help?> Profile.print
...
groupby – Controls grouping over tasks and threads, or no grouping. Options are :none (default), :thread, :task, [:thread, :task], or [:task, :thread] where the last two provide
       nested grouping.
```

And by [ProfileView.jl](https://github.com/timholy/ProfileView.jl/), and [ProfileCanvas.jl](https://github.com/pfitzseb/ProfileCanvas.jl) (which is what i think the VS Code profile view uses)